adafruit_rfm/rfm9xfsk.py

```diff
@@ -0,0 +1,572 @@
+
+    @property
+    def rssi(self) -> float:
+        """The received strength indicator (in dBm) of the last received message."""
+        # Read RSSI register and convert to value using formula in datasheet.
+        # Remember in LoRa mode the payload register changes function to RSSI!
+        raw_rssi = self.read_u8(_RF95_REG_11_RSSI_VALUE)
+        return -raw_rssi / 2.0
+
+    @property
+    def enable_crc(self) -> bool:
+        """Set to True to enable hardware CRC checking of incoming packets.
+        Incoming packets that fail the CRC check are not processed.  Set to
+        False to disable CRC checking and process all incoming packets."""
+        return self.crc_on
+
+    @enable_crc.setter
+    def enable_crc(self, val: bool) -> None:
+        # Optionally enable CRC checking on incoming packets.
+        if val:
+            self.crc_on = 1
+            self.crc_type = 0  # use CCITT for RF69 compatibility
+        else:
+            self.crc_on = 0
+
+    def crc_error(self) -> bool:
+        """crc status"""
+        return (self.read_u8(_RF95_REG_3F_IRQ_FLAGS_2) & 0x2) >> 1
+
+    @property
+    def enable_address_filter(self) -> bool:
+        """Set to True to enable address filtering.
+        Incoming packets that do no match the node address  or broadcast address
+        will be ignored."""
+        return self.address_filter
+
+    @enable_address_filter.setter
+    def enable_address_filter(self, val: bool) -> None:
+        # Enable address filtering  on incoming packets.
+        if val:
+            self.address_filter = 2  # accept node address or broadcast address
+        else:
+            self.address_filter = 0
+
+    @property
+    def fsk_node_address(self) -> int:
+        """Node Address for Address Filtering"""
+        return self.read_u8(_RF95_REG_33_NODE_ADDR)
+
+    @fsk_node_address.setter
+    def fsk_node_address(self, val: int) -> None:
+        assert 0 <= val <= 255
+        self.write_u8(_RF95_REG_33_NODE_ADDR, val)
+
+    @property
+    def fsk_broadcast_address(self) -> int:
+        """Node Address for Address Filtering"""
+        return self.read_u8(_RF95_REG_34_BROADCAST_ADDR)
+
+    @fsk_broadcast_address.setter
+    def fsk_broadcast_address(self, val: int) -> None:
+        assert 0 <= val <= 255
+        self.write_u8(_RF95_REG_34_BROADCAST_ADDR, val)
+
+    @property
+    def ook_fixed_threshold(self) -> int:
+        """Fixed threshold for data slicer in OOK mode"""
+        return self.read_u8(_RF95_REG_15_OOK_FIX)
+
+    @ook_fixed_threshold.setter
+    def ook_fixed_threshold(self, val: int) -> None:
+        assert 0 <= val <= 255
+        self.write_u8(_RF95_REG_15_OOK_FIX, val)
+
+    def packet_sent(self) -> bool:
+        """Transmit status"""
+        return (self.read_u8(_RF95_REG_3F_IRQ_FLAGS_2) & 0x8) >> 3
+
+    def payload_ready(self) -> bool:
+        """Receive status"""
+        return (self.read_u8(_RF95_REG_3F_IRQ_FLAGS_2) & 0x4) >> 2
```

Comment:
Should these be properties too?

Reply:
Yes, I thinks so. Will change.

Reply:
Hmm -- I think these do have to remain as callable functions since they are passed into asyncio_check_timeout and check repeatedly there. We don opt wasn't to just pass in the value at the time of the initial call.